src/network/wait-online/link.c

```diff
@@ -145,6 +152,90 @@ static int link_update_altnames(Link *l, sd_netlink_message *m) {
         return 0;
 }
 
+static int link_update_dns_from_resolv_conf(Link *l) {
+        _cleanup_strv_free_ char **dns = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+
+        log_link_debug(l, "Attempting to read DNS servers from /etc/resolv.conf");
+
+        f = fopen("/etc/resolv.conf", "re");
+        if (!f)
+                return -errno;
```

Comment:
please log about this at debug level, i.e. turn this line into `return log_debug_errno(errno, "…` with some vaguely useful log message, this is otherwise too hard to debug if there are issues anywhere